### PR TITLE
More types support

### LINKED ISSUE #89: Show TypeLambda-s in signatures properly
Example of a fun `TypeLambda`:

```scala
type F = [X >: 5 <: AnyRef, Y] => [Z] => (Option[X], Y => Unit, Z)
```

### LINKED ISSUE #92: Show Type/Term-Refs with non-trivial qualifiers in signatures

### LINKED ISSUE #93: Show polyfunction types in signatures
Example:

```scala
type F = [X] => (x: X) => List[X]
```

Note that this isn't the same as a type lambda, and vice versa.

### PR DESCRIPTION
**Pull request by pikinier20**
closes #89 closes #92 closes #93

### CODE REVIEW

File: src/main/scala/tests/typesSignatures.scala [hunk 1]
```diff
@@ -4,8 +4,12 @@ package typesSignatures
...
 {
   type A = Int
-  type B[+T] = Seq[T]
-  type C[A, B <: A] = Seq[B]
+  // type B[+T] = Seq[T] It's translated into:
+  type B = [T] =>> Seq[T]
+  //variance is checked by compiler and info about it has no use for developer 
+  
+  //type C[A, B <: A] = Seq[B] Equivalent form:
+  type C = [A, B <: A] =>> Seq[B]
```
Comment (abgruszecki) [inline src/main/scala/tests/typesSignatures.scala]: Can we keep the old tests? If we don't, we risk regressing on them.

**Review by abgruszecki — COMMENTED** [review]
Looks good, just a small fix for the tests would be good.

File: src/main/scala/tests/typesSignatures.scala (hunk 1 above)
Comment (pikinier20) [reply src/main/scala/tests/typesSignatures.scala]: There's no information about how `type C` is defined in source file (I mean if user used shorter syntax or type lambda). In TASTY we always see it as TypeLambda. I needed to decide how I want to show it. I know that this case could be shown in shorter form, but the decision how I parsed type tree applies to many other cases and I didn't want to corrupt them.

Comment (pikinier20) [reply src/main/scala/tests/typesSignatures.scala]: The rule I used is that TypeLambda for abstract/opaque type is represented in shorter form, but for regular types it's shown as TypeLambda

Comment (abgruszecki) [reply src/main/scala/tests/typesSignatures.scala]: Hm, that's correct - I thought Dotty REPL could distinguish between the two, but it cannot do that as well. In that case I'd actually do what the REPL does and always display type members in the `type F[+A] = List[A]` form. We can test type lambda rendering by creating aliases like

```scala
type Id[T] = T
type G = Id[[A] =>> List[A]]
```

Comment (pikinier20) [reply src/main/scala/tests/typesSignatures.scala]: Ok, done.

**Pull request merged**